logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/applock/AppLockTeamFeatureConfigObserver.kt

```diff
@@ -31,22 +34,31 @@ interface AppLockTeamFeatureConfigObserver {
 }
 
 class AppLockTeamFeatureConfigObserverImpl(
-    private val userConfigRepository: UserConfigRepository
+    private val userConfigRepository: UserConfigRepository,
+    private val kaliumConfigs: KaliumConfigs
 ) : AppLockTeamFeatureConfigObserver {
-    override fun invoke(): Flow<AppLockTeamConfig> =
-        userConfigRepository.observeAppLockStatus().map {
-            it.fold({
+    override fun invoke(): Flow<AppLockTeamConfig> {
+        if (kaliumConfigs.teamAppLock) {
+            return flowOf(
                 AppLockTeamConfig(
-                    isEnabled = false,
-                    timeout = DEFAULT_TIMEOUT
+                    isEnabled = true,
+                    timeout = kaliumConfigs.teamAppLockTimeout.seconds,
+                    isStatusChanged = false
                 )
-            }, { appLockModel ->
+            )
+        }
+        return userConfigRepository.observeAppLockConfig().map {
+            it.fold({
```

Comment:
you can use `mapLeft` instead

Reply:
mapLeft returns Either, in my case I want to return AppLockTeamConfig object